exporters/failure/collector_azure_devops.py

```diff
@@ -0,0 +1,174 @@
+# Copyright Red Hat
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+#
+
+import logging
+from typing import List
+
+from attrs import converters, define, field
+from azure.devops.connection import Connection
+from azure.devops.exceptions import AzureDevOpsServiceError
+from azure.devops.v6_0.work_item_tracking.models import Wiql, WorkItem
+from msrest.authentication import BasicAuthentication
+
+from failure.collector_base import AbstractFailureCollector, TrackerIssue
+from pelorus.config import env_var_names, env_vars
+from pelorus.config.converters import comma_or_whitespace_separated, pass_through
+from pelorus.config.log import REDACT, log
+from pelorus.errors import FailureProviderAuthenticationError
+from pelorus.timeutil import parse_assuming_utc, second_precision
+from pelorus.utils import Url
+
+_DATETIME_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
+
+
+@define(kw_only=True)
+class AzureDevOpsFailureCollector(AbstractFailureCollector):
+    """
+    Azure DevOps implementation of a FailureCollector
+    """
+
+    token: str = field(
+        metadata=env_vars(*env_var_names.TOKEN) | log(REDACT),
+        repr=False,
+    )
+
+    tracker_api: Url = field(
+        metadata=env_vars("SERVER"),
+        converter=converters.optional(pass_through(Url, Url.parse)),
+    )
+
+    projects: set[str] = field(
+        factory=set, converter=comma_or_whitespace_separated(set)
+    )
+
+    work_item_type: set[str] = field(
+        factory=set,
+        converter=comma_or_whitespace_separated(set),
+        metadata=env_vars("AZURE_DEVOPS_TYPE"),
+    )
+
+    work_item_priority: set[str] = field(
+        factory=set,
+        converter=comma_or_whitespace_separated(set),
+        metadata=env_vars("AZURE_DEVOPS_PRIORITY"),
+    )
+
+    def __attrs_post_init__(self):
+        credentials = BasicAuthentication("", self.token)
+        connection = Connection(base_url=self.tracker_api.url, creds=credentials)
+        self.client = connection.clients_v6_0.get_work_item_tracking_client()
+
+    def get_work_items(self) -> List[WorkItem]:
+        logging.info("Getting work items")
+
+        try:
+            wiql = Wiql(query="Select [System.Id] From WorkItems")
```

Comment:
In JIRA we have an option of custom query, should this be similar here?

Reply:
I think it would work, but do you believe it is necessary?

Reply:
I don't know, does the user can change the workflow of the azure devops tickets? If yes then that may be a feature request (separate PR) if not then it's ok as it is.